pkg/core/ratelimits/matcher.go

```diff
@@ -42,18 +42,32 @@ func buildRateLimitMap(
 
 	additionalInbounds, err := manager_dataplane.AdditionalInbounds(dataplane, mesh)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not fetch additional inbounds")
+		return core_xds.RateLimitsMap{}, errors.Wrap(err, "could not fetch additional inbounds")
 	}
 	inbounds := append(dataplane.Spec.GetNetworking().GetInbound(), additionalInbounds...)
 	policyMap := policy.SelectInboundConnectionMatchingPolicies(dataplane, inbounds, policies)
 
-	result := core_xds.RateLimitsMap{}
+	result := core_xds.RateLimitsMap{
+		Inbound:  core_xds.InboundRateLimitsMap{},
+		Outbound: core_xds.OutboundRateLimitsMap{},
+	}
 	for inbound, connectionPolicies := range policyMap {
-		result[inbound] = []*mesh_proto.RateLimit{}
+		result.Inbound[inbound] = []*mesh_proto.RateLimit{}
 		for _, policy := range connectionPolicies {
-			result[inbound] = append(result[inbound], policy.(*mesh_core.RateLimitResource).Spec)
+			result.Inbound[inbound] = append(result.Inbound[inbound], policy.(*mesh_core.RateLimitResource).Spec)
+		}
+	}
+
+	outboundMap := policy.SelectOutboundConnectionPolicies(dataplane, policies)
+
+	for _, outbound := range dataplane.Spec.Networking.GetOutbound() {
+		serviceName := outbound.GetTagsIncludingLegacy()[mesh_proto.ServiceTag]
+		oface := dataplane.Spec.GetNetworking().ToOutboundInterface(outbound)
+		if policy, exists := outboundMap[serviceName]; exists {
+			result.Outbound[oface] = append(result.Outbound[oface], policy.(*mesh_core.RateLimitResource).Spec)
```

Comment:
An outbound interface is a address:port, so isn't that unique? That is, there is one outbound interface per outbound? In which case there's only one rate limit per outbound interface?

Could be I'm missing something :)

Reply:
It seems like we need `SelectOutboundConnectionAllPolicies` that picks all matching policies for every outbound

Reply:
Yes @lobkovilya this actually needs them sorted in the same way that inbound rate limit routes are sorted.

Reply:
@jpeach , @lobkovilya and I discussed this today and finally resolved how to handle this - we do not support multiple, indistinguishable services on a single outbound. When looking at an individual envoy outbound here, we support exactly one pair of source and dest tags, so we only need to match the most specific policy to those tags, which is provided by `SelectOutboundConnectionPolicies`. So, `result.Outbound` will hold a single policy rather than an array, and that policy will be the only filter applied. TLDR - your original comment was correct :)